crates/wit/src/commands/pull.rs

```diff
@@ -0,0 +1,273 @@
+use std::{
+    collections::{BTreeSet, HashSet},
+    io::Write,
+    path::PathBuf,
+};
+
+use anyhow::{bail, Context, Result};
+use clap::Args;
+
+use cargo_component_core::{command::CommonOptions, VersionedPackageName};
+use futures::TryStreamExt;
+use tokio_util::io::{StreamReader, SyncIoBridge};
+use warg_loader::{ClientConfig, Release};
+use wit_component::DecodedWasm;
+use wit_parser::{PackageId, PackageName, Resolve, UnresolvedPackage};
+
+/// Pull WIT package(s) to a local "deps" directory.
+#[derive(Args)]
+#[clap(disable_version_flag = true)]
+pub struct PullCommand {
+    /// The common command options.
+    #[clap(flatten)]
+    pub common: CommonOptions,
+
+    /// Use the specified registry name when pulling the package(s).
+    #[clap(long, value_name = "REGISTRY")]
+    pub registry: Option<String>,
+
+    /// Update the specified directory WIT "root" directory. Dependencies will
+    /// be written to e.g. "<wit-dir>/deps/<namespace>.<package>.wit".
+    #[clap(long, value_name = "WIT_DIR", default_value = "wit")]
+    pub wit_dir: PathBuf,
+
+    /// Create "<wit-dir>" and "<wit-dir>/deps" directories if needed.
```

Comment:
I guess this would create wit-dir if it doesn't exist, but as I indicate below, I think that's an error condition and this should only be used to create the `deps` subdirectory if it doesn't exist.

Reply:
I was thinking about `--create-dirs` in terms of initing a new project, but I suppose the chance of accidentally doing this in the wrong parent dir probably outweighs the tiny inconvenience of creating one dir.

I think I'll get rid of the option and just create the `deps` dir unconditionally when needed.